py/desispec/zcatalog.py

```diff
@@ -232,14 +249,33 @@ def create_summary_catalog(specprod, specgroup = 'zpix', \
     ## Adding all the tables into a single list
     tables = []
 
+    ## Handle DEPNAMnn DEPVERnn keyword merging separately
+    dependencies = dict()
+
     ## Looping through the different zcatalogs and adding the survey and program columns
     for filename in zcat:
         basefile = os.path.basename(filename)
 
         ## Load the ZCATALOG table, along with the meta data
-        log.debug(f'Reading {filename}')
+        log.info(f'Reading {filename}')
         t = Table.read(filename, hdu = 'ZCATALOG')
 
+        ## Merge DEPNAMnn and DEPVERnn, then remove from header
+        desiutil.depend.mergedep(t.meta, dependencies)
```

Comment:
It looks like this code is exactly duplicated in the `collect_dependencies` function in `bin/desi_zcatalog`. Why does identical code need to be in two different places?

Reply:
good point; desihub/desiutil#202 will add functions to desiutil.depend to support removing dependencies to avoid the repeated code here.  I'll update this PR after that is merged.